src/Illuminate/Routing/UrlGenerator.php

```diff
@@ -671,7 +671,7 @@ public function format($root, $path, $route = null)
      */
     public function isValidUrl($path)
     {
-        if (! preg_match('~^(#|//|https?://|(mailto|tel|sms):)~', $path)) {
+        if (! preg_match('~^(#|//|https?://|(mailto|tel|sms):)~', (string) $path)) {
```

Comment:
This is also the wrong place to put this. It needs to go in the **caller** of `isValidUrl` not inside the method body. The place that calls `isValidUrl` needs to make sure to only pass a string.